File: IOS/Sources/OpenRoaming/OpenRoaming/Services/DeleteProfileService.swift
//
//  DeleteProfileService.swift
//  OpenRoaming
//
//  Created by olivier duque campos on 23/06/20.
//  Copyright © 2020 Cisco. All rights reserved.
//

import Foundation

class DeleteProfileService: Service {
    
    /**
    * Method responsible to delete a profile on openroaming backend.
    */
    override func onStart(headerParams: Array<HeaderParam>, serviceHandler: @escaping ServiceHandler) {
        let urlString = Urls.userDeleteProfileAccountUrl()
        let requestData = RequestData(url: urlString, headerParams: headerParams, method: HTTPMethod.DELETE)
        call(requestData: requestData, serviceHandler: serviceHandler)
    }
}

File: Android/src/main/java/com/cisco/or/sdk/services/DeleteProfileService.kt
package com.cisco.or.sdk.services

import android.content.Context
import com.cisco.or.sdk.enums.IdType
import com.cisco.or.sdk.types.ServiceHandler
import com.cisco.or.sdk.utils.Urls
import com.cisco.or.sdk.utils.Utils
import org.json.JSONObject
import java.util.*
import kotlin.collections.ArrayList

/**
 * Delete Profile on openroaming backend
 */
internal object DeleteProfileService : Service() {
    private val TAG: String = DeleteProfileService::class.java.name

    override fun onStart(context: Context, headerParams: ArrayList<HeaderParam>, serviceHandler: ServiceHandler) {

        val url = Urls.deleteProfileUrl()
        val requestData = RequestData(url, headerParams, HTTP_METHOD.DELETE, null)
        call(context, requestData, RESPONSE_FORMATTER.TEXT, serviceHandler)
    }
}